cmd/guaccollect/cmd/files.go

```diff
@@ -113,7 +113,7 @@ func getCollectorPublish(ctx context.Context, blobStore *blob.BlobStore, pubsub
 	}, nil
 }
 
-func initializeNATsandCollector(ctx context.Context, pubsubAddr string, blobAddr string) {
+func initializeNATsandCollector(ctx context.Context, pubsubAddr string, blobAddr string, isPoll bool) {
```

Comment:
I wonder if, for the sake of future-proofing, is poll can be moved to an integer to indicate the polling interval. Currently as things stand, `g` holds two flags: 1) poll (yes/no) and 2) interval (numeric) that are tightly coupled (i.e., interval should never be another value unless poll is true). 

Not sure if this make sense!

Reply:
I am not sure if I understand you correctly, please correct me if I am wrong.

My understanding of your comment is that we should change `isPoll` to the actual poll interval. If the poll interval is `0` we don’t poll, and if it is any other positive value the we know that we have to poll.

Reply:
Essentially, this way you don't need to bookkeep two different variables and reconcile when they are conflicting (i.e., choose a winner).

Reply:
Thanks @SantiagoTorres!

After spending some time thinking about this, I realized that replacing the two flags (`poll` and `interval`) with just the `interval` flag would make it confusing for other developers trying to understand the code.

My reasoning is that it will become confusing for people who want to edit the code because they won’t actually know that polling is `false` when `interval` is equal to zero. 

Even if we add comments to state this, most people ignore comments and they won’t know why their polling is messing up because there would be no actual `poll` flag.